compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1159,6 +1158,7 @@ void OMR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
         srcOffHigh >= 0 && dstOffHigh >= 0 && copyLenHigh >= 0)
       {
       transformTheCall = true;
```

Comment:
I think the original `fullTransform` means that the transformation can be done even without any type information here. Previously it just so happened that `primitiveTransform` implied `fullTransform`, so the `primitiveTransform || fullTransform` condition actually guaranteed `fullTransform`. In that case, `primitiveTransform && referenceTransform` seems to be the equivalent of the old `fullTransform`.

I agree that larger changes to the logic here are outside the scope of this change. However, changing the default away from `true` is safe because it can only make the transformation more conservative. The default I suggested above wouldn't affect performance either, because it's still the case (for now) that all platforms have both `primitiveTransform` and `referenceTransform`.

By the way, I've verified that your changes to the logic below do not affect the resulting values of `transformTheCall` and `needArrayCheck`, given the current and previous values on each platform of `primitiveTransform` and `fullTransform`/`referenceTransform`. I checked each (possible) combination of values of the following:
* `primitiveArray1`,
* `primitiveArray2`,
* `referenceArray1`,
* `referenceArray2`,
* `identicalTypes`, standing in for `[...] && srcObject...->getClass() == dstObject...->getClass()`, and
* `alwaysCallWriteBarrier`, standing in for `comp()->getOptions()->alwaysCallWriteBarrier()`.